internal/discord/commands.go

```diff
@@ -112,7 +112,153 @@ func (d *Discord) Purge(s *discordgo.Session, i *discordgo.InteractionCreate) {
+		return
+	}
+
+	// Check if user already has exile role
+	isExiled := false
+	for _, role := range memberToExile.Roles {
+		if role == exileRole.ID {
+			isExiled = true
+		}
+	}
+	if isExiled {
+		tempstr := fmt.Sprintf("User <@%v> is already exiled", userToExile.ID)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
+
+	// Exile the user
+	// Remove Verified role
+	err = s.GuildMemberRoleRemove(i.GuildID, userToExile.ID, verifiedRole.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not remove role %v from user <@%v>", verifiedRole.Name, userToExile.ID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
+	// Add exiled role
+	err = s.GuildMemberRoleAdd(i.GuildID, userToExile.ID, exileRole.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not give user <@%v> role %v", userToExile.ID, exileRole.Name)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
+
+	// Inform invoker and edit log message of successful exile
+	endTime := startTime.Add(duration)
+	tempstr := fmt.Sprintf(
+		"\nUser <@%v> has been exiled until <t:%v>",
+		userToExile.ID,
+		endTime.Unix(),
+	)
+	logMsg.Embeds[0].Description += tempstr
+	RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+
+	// DM user regarding the exile
+	channel, err := s.UserChannelCreate(userToExile.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not create a DM with user %v", userToExile.ID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		logMsg.Embeds[0].Description += "\nFailed to notify of the exile via DM"
+	} else {
+		tempstr := fmt.Sprintf("You are being exiled from `%v` until <t:%v> for the following reason:\n> %v",
+			GuildName,
+			endTime.Unix(),
+			optionMap["reason"].StringValue(),
+		)
+		_, err = s.ChannelMessageSend(channel.ID, tempstr)
+		if err != nil {
+			tempstr := fmt.Sprintf("Could not send a DM to user <@%v>", userToExile.ID)
+			fmt.Printf("%v: %v\n", tempstr, err)
+			RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+			logMsg.Embeds[0].Description += "\nFailed to notify of the exile via DM"
+		}
+	}
```

Comment:
we can make a general function for dming someone since we'll be doing this in three commands so far, `Exile`, `Unexile`, and `Kick`. we can pass in a string and have it send the DM